sopel/bot.py

```diff
@@ -17,7 +17,9 @@
 from types import MappingProxyType
 from typing import (
     Any,
+    Callable,
     Optional,
+    Sequence,
```

Comment:
Any reason not to import these from `collections.abc` since the `typing` counterparts are deprecated since Python 3.9?

Reply:
The only reason would be "not touching more than necessary", I think.

Switching from the deprecated imports to their replacements is unrelated to the PR's goal of making `--disallow-incomplete-defs` work.